saleor/graphql/attribute/utils.py

```diff
@@ -370,20 +371,36 @@ def save(cls, instance: T_INSTANCE, cleaned_input: T_INPUT_MAP):
         :param instance: the product or variant to associate the attribute against.
         :param cleaned_input: the cleaned user input (refer to clean_attributes)
         """
+        errors = []
+
         for attribute, attr_values in cleaned_input:
-            if attribute.input_type == AttributeInputType.FILE:
-                attribute_values = cls._pre_save_file_value(
-                    instance, attribute, attr_values
+            try:
+                if attribute.input_type == AttributeInputType.FILE:
+                    attribute_values = cls._pre_save_file_value(
+                        instance, attribute, attr_values
+                    )
+                elif attribute.input_type == AttributeInputType.REFERENCE:
+                    attribute_values = cls._pre_save_reference_values(
+                        instance, attribute, attr_values
+                    )
+                else:
+                    attribute_values = cls._pre_save_values(attribute, attr_values)
+
+                associate_attribute_values_to_instance(
+                    instance, attribute, *attribute_values
                 )
-            elif attribute.input_type == AttributeInputType.REFERENCE:
-                attribute_values = cls._pre_save_reference_values(
-                    instance, attribute, attr_values
+
+            except DataError as e:
```

Comment:
It can be thrown, if some1 passes string larger than the current `name` max_length.